packages/cli/src/telemetry/index.ts

```diff
@@ -171,8 +171,13 @@ export class Telemetry {
 
 	async trackN8nStop(): Promise<void> {
 		clearInterval(this.pulseIntervalReference);
-		await this.track('User instance stopped');
-		void Promise.all([this.postHog.stop(), this.rudderStack?.flush()]);
+
+		// Send the stopped event and make sure it gets flushed right away
+		await Promise.all([
```

Comment:
Promise.all does not guarantee a sequential execution. the track call might happen before the flush.

Reply:
No it doesn't, and here it doesn't need to. The track call pushes to the queue synchronously, and returns a promise that resolves when the event is sent. This happens synchronously before the flush call is executed. The flush then sends the event and any other events in the queue. Very confusing yes.

I could have written it like this:

```ts
void this.track()

await Promise.all([this.postHog.stop(), this.rudderStack?.flush()]
```

But I noticed that RS's `flush` only [sends at max 20 events at a time](https://github.com/rudderlabs/rudder-sdk-node/blob/master/src/index.js#L625) (unless `flushAt` is [changed](https://www.rudderstack.com/docs/sources/event-streams/sdks/rudderstack-node-sdk/#sdk-initialization-options)). So theoretically if there are more than 20 events in the queue it might not get sent with the first flush. Altho looking at the code again it might be that that scenario never happens.

I think we could actually simplify by making the `track` function entirely synchronous, as I couldn't find any cases where it would actually make sense to wait until the event has been sent. And if that is actually the case, one can always call `flush`. WDYT?

Reply:
> I think we could actually simplify by making the `track` function entirely synchronous

Yes please 🙏🏽

Reply:
@netroy refactor all the `track` and `identify` calls to be synchronous. Please have a look 🙏